### ISSUE TITLE
Crash when transitioning away from a page with ui-material-tabs

### ISSUE THREAD

**Issue by cjohn001** [nativescript-community/ui-material-components#327]
### Which platform(s) does your issue occur on?
- iOS only
- iOS only 14.5 tested 
- emulator and device. 

### Please, provide the following version numbers that your issue occurs with:

- CLI: 8.0.2-next-08-13-2021-1128363962
- Cross-platform modules: 8.1.0-sap.0
`node_modules/@nativescript/core/package.json` file in your project)
- Runtime(s): JSC and rc
- Plugin(s): I tried with
-ui-material-tabs": "5.3.19" and "5.3.15"

"devDependencies": {
		"@angular/compiler-cli": "~12.2.2",
		"@graphql-codegen/cli": "^1.20.1",
		"@graphql-codegen/fragment-matcher": "^2.0.1",
		"@graphql-codegen/introspection": "^1.18.1",
		"@graphql-codegen/typescript": "^1.21.1",
		"@graphql-codegen/typescript-apollo-angular": "~2.3.1",
		"@graphql-codegen/typescript-operations": "^1.17.15",
		"@nativescript/android": "8.0.0",
		"@nativescript/ios": "JSC",
		"@nativescript/types": "~8.0.1",
		"@nativescript/webpack": "^5.0.0-rc.4",
		"@ngtools/webpack": "~12.2.0",
		"@types/d3-ease": "^2.0.0",
		"@types/intl": "^1.2.0",
		"@types/node": "^13.13.19",
		"@types/uuid": "^8.3.0",
		"codelyzer": "^6.0.1",
		"keycloak-request-token": "^0.1.0",
		"sass": "^1.37.5",
		"ts-node": "^9.1.1",
		"tslint": "^6.1.3",
		"typescript": "~4.2.0"
	},

### Please, tell us how to recreate the issue in as much detail as possible. 
Unfortunately, the bug seems to be not easy to reproduce. It seems to be a runtime issue which not always appears. The problem: I have a page with the material tabs control including 4 tab pages. Everything is working fine. When I try to navigate away to a new page I randomly see the following error message without a stack trace.

NativeScript discarding uncaught JS exception!

The JSC runtime does not brake but just seems to ignore the issue. With the new v8 runtime the app crashes instead.

The just described behavior appears with "5.3.19" of the plugin. I downgraded to "5.3.15" where I see the same bug. But here I got an additional stacktrace which is given below:

CONSOLE LOG file: node_modules/@angular/core/fesm2015/core.js:28039:0 Angular is running in development mode. Call enableProdMode() to enable production mode.
NativeScript discarding uncaught JS exception!
NativeScript discarding uncaught JS exception!
***** Fatal JavaScript exception - application has been terminated. *****
Native stack trace:
1   0x10840d62e NativeScript::reportFatalErrorBeforeShutdown(JSC::ExecState*, JSC::Exception*, bool)
2   0x10845f524 -[TNSRuntime executeModule:referredBy:]
3   0x107645093 main
4   0x7fff2025abbd start
JavaScript stack trace:
UIApplicationMain(file: node_modules/@nativescript/core/application/index.ios.js:336:0
at run(file: node_modules/@nativescript/core/application/index.ios.js:336:0
at runNativeScriptAngularApp(file: node_modules/@nativescript/angular/fesm2015/nativescript-angular.js:564:19
at ./src/main.ts(file: src/src/main.ts:23:25
at __webpack_require__(file: src/webpack/bootstrap:19:0
at __webpack_exec__(file:///app/bundle.js:36557:58)
at file:///app/bundle.js:36558:127
at file: src/webpack/runtime/startup entrypoint:6:0
at file:///app/bundle.js:36558:48
at anonymous(file:///app/bundle.js:36563:3)
at evaluate([native code])
at moduleEvaluation([native code])
at [native code]
at asyncFunctionResume([native code])
at [native code]
at promiseReactionJob([native code])
JavaScript error:
file: node_modules/@nativescript/core/application/index.ios.js:336:0 JS ERROR Error: Failed to allocate data stores for 140362783383168 rows in section 0. Consider using fewer rows

@farfromrefug : I know it is very unlikely that you will find the issue with the given details. I am wondering, if Nativescript has some kind of tracing functionality which I could activate to get more information like a stacktrace for the most recent plugin version. It seems that there is some uninitialzed variable involved (rows?).

**Comment by farfromrefug** [comment]
@cjohn001 thanks i know understand the issue. The view controller try to layout its subviews after the tabbar have  been released. It should be easy to fix. 
I wonder if this is another issue though as it is not the same error message. 
Will fix tonight or tomorrow.

**Comment by cjohn001** [comment]
@farfromrefug: The error with the tabbar is the one which I observed in the latest version of the plugin
-ui-material-tabs": "5.3.19" 
As I could only see the single line error message: NativeScript discarding uncaught JS exception!

I downgraded to "5.3.15". From which. the stacktrace with the rows was obtained from. I will further have an eye on the topic if I see the issue from the old plugin again.
I indeed have another issue with the UICollectionView. But it is not related to the issue here, therefore I will provide an extra issue in its repository later the day. Need to prepare some stuff first in order to be able to show the problem.
